arraycontext/fake_numpy.py

```diff
@@ -91,25 +91,20 @@ def _get_fake_numpy_linalg_namespace(self):
         # "interp",
         })
 
-    def _new_like(self, ary, alloc_like):
-        if np.isscalar(ary):
-            # NOTE: `np.zeros_like(x)` returns `array(x, shape=())`, which
-            # is best implemented by concrete array contexts, if at all
-            raise NotImplementedError("operation not implemented for scalars")
-
+    def empty_like(self, ary):
         if isinstance(ary, np.ndarray) and ary.dtype.char == "O":
             # NOTE: we don't want to match numpy semantics on object arrays,
-            # e.g. `np.zeros_like(x)` returns `array([0, 0, ...], dtype=object)`
+            # e.g. `np.empty_like(ary)` returns `array([None, ...], dtype=object)`
             # FIXME: what about object arrays nested in an ArrayContainer?
             raise NotImplementedError("operation not implemented for object arrays")
 
-        return rec_map_array_container(alloc_like, ary)
-
-    def empty_like(self, ary):
-        return self._new_like(ary, self._array_context.empty_like)
+        return self._array_context.empty_like(ary)
 
     def zeros_like(self, ary):
```

Comment:
This makes me wonder whether we should keep both `actx.np.zeros_like` or `actx.zeros_like` indefinitely. I'm leaning towards deprecating the `actx` version. What do you think?

Reply:
Yeah, agreed. 

`actx.np.zeros_like` should definitely be there to match `numpy`, which makes `actx.zeros_like` a bit redundant now that it works with array containers.